packages/runtime/src/extensibility/facades/consumption/AttributesFacade.ts

```diff
@@ -14,6 +14,8 @@ import {
     DeletePeerSharedAttributeAndNotifyOwnerUseCase,
     DeleteRepositoryAttributeRequest,
     DeleteRepositoryAttributeUseCase,
+    DeleteSharedAttributesForRejectedOrRevokedRelationshipRequest,
+    DeleteSharedAttributesForRejectedOrRevokedRelationshipUseCase,
```

Comment:
Would you kindly explain your reasoning behind making this a use case instead of automatically deleting the data? The data left behind after a revoked or rejected relationship request feels of little use to me and I'd rather have it removed immediately. Automating this would mean less data trash and one less use case.

Sorry if you guys already though/spoke about this. I was gone most of last week.

Reply:
sure! this has to be done by a module, modules are pieces of code that automate things that runtime users **could** do themselves (when not enabling the module). Therefore modules only have access to things that are coming out of the extensibility -> we have to expose the functionality.

As soon as we don't expose this to App / Connector we should be very safe!

Reply:
A different approach: how about we use a private, internal module that the user has no access to and into which we @Inject the attributes controller?

Reply:
There is no `the attribute controller` b/c we have to be prepared for multi accounting here. Therefore this is not possible.

Reply:
And also not required. I don't see any problems with the current approach. It's a widely used pattern to create usecases for things that are only used in our modules.